src/Terrain/TerrainType.java

```diff
@@ -161,9 +163,10 @@ private TerrainType(int attributeFlags, int defense, Color mainColor, String nam
   public static final TerrainType SHOAL = new TerrainType( SHOAL_FLAGS, SHOAL_DEFENSE, SHOAL_COLOR, SHOAL_NAME );
 
   // List of all terrain types.
-  public static final TerrainType[] TerrainTypeList = {
-    AIRPORT, BRIDGE, CITY, BUNKER, PILLAR, DUNES, FACTORY, FOREST, GRASS, HEADQUARTERS, LAB, MOUNTAIN, REEF, RIVER, ROAD, SEA, SEAPORT, SHOAL
-  };
+  public static final ArrayList<TerrainType> TerrainTypeList =
```

Comment:
Why change this to an `ArrayList`?

Reply:
Easier to work with. I didn't know how before.

Reply:
Huh. I don't think I see any new (kinds of) interactions with `TerrainTypeList` that weren't there before.
I mean, I guess it works either way, but I kinda prefer the simplicity of the standard array when the size is known a priori and it will never change.

Reply:
`IndexOf()`, for putting the index of the terrain type out into a file without having to write a helper function, haha.